### PR TITLE
Provide blueocean link as an input to RunTestPlan command

### PR DESCRIPTION
**Pull request by sameerawickramasekara**
## Purpose
To provide quick access to the blue ocean dashboard for each build we generate the link in pipeline and provide it to RunTestPlan command to store in the database.

### **IMPORTANT**
**Merge this after merging the related TestGrid PR to avoid any build breaks**

### CODE REVIEW

File: src/org/wso2/tg/jenkins/executors/TestExecutor.groovy [hunk 1]
```diff
@@ -162,6 +172,39 @@ def prepareWorkspace(testPlanId, scenarioConfigs) {
...
+                    if(node.displayName.endsWith(parallelname)){
+                        println("Parallel node  : " + parallelname + " : " + node.id.toString())
+                        parallelId = node.id.toString()
+                    }
+                }
+
+            }
+        }
+    }
+    build = null
```
Comment (yasassri) [inline src/org/wso2/tg/jenkins/executors/TestExecutor.groovy]: Shall we add a  comment why you are doing this?

File: src/org/wso2/tg/jenkins/executors/TestExecutor.groovy [hunk 2]
```diff
@@ -228,3 +271,21 @@ static def getRepositoryBranch(def branch) {
...
+ * Builds the bluocean console link for a given test plan
+ *
+ * @param buildURL Jenkins classic build URL
+ * @param parallelId Parallel id of the test plan
+ * @return Blueocean link
+ */
+def generateBluoceanLink(def buildURL , def parallelId){
+    def split = buildURL.split('/')
+    def url = null
+    if (split.length > 0 ){
```
Comment (yasassri) [inline src/org/wso2/tg/jenkins/executors/TestExecutor.groovy]: Should we do split.length > 5 ?

File: src/org/wso2/tg/jenkins/executors/TestGridExecutor.groovy [hunk 3]
```diff
@@ -47,15 +47,15 @@ def generateTesPlans(def product, def configYaml) {
  * @param testPlanFilePath test plan location
  * @param workspace execution workspace
```
Comment (yasassri) [inline src/org/wso2/tg/jenkins/executors/TestGridExecutor.groovy]: Missing Javadoc

File: src/org/wso2/tg/jenkins/executors/TestGridExecutor.groovy [hunk 4]
```diff
@@ -47,15 +47,15 @@ def generateTesPlans(def product, def configYaml) {
...
  */
-def runTesPlans(def product, def testPlanFilePath, def workspace) {
+def runTesPlans(def product, def testPlanFilePath, def workspace, def url) {
```
Comment (yasassri) [inline src/org/wso2/tg/jenkins/executors/TestGridExecutor.groovy]: Shall we pdate the javadoc?

File: src/org/wso2/tg/jenkins/util/Common.groovy [hunk 5]
```diff
@@ -19,6 +19,8 @@
 package org.wso2.tg.jenkins.util
 
 import org.wso2.tg.jenkins.Properties
+import org.wso2.tg.jenkins.Logger
```
Comment (yasassri) [inline src/org/wso2/tg/jenkins/util/Common.groovy]: Is this used?

File: src/org/wso2/tg/jenkins/executors/TestExecutor.groovy [hunk 6]
```diff
@@ -37,8 +40,13 @@ def runPlan(tPlan, testPlanId) {
     def log = new Logger()
     def scenarioConfigs = []
 
+    def execName = commonUtil.extractInfraCombination("${testPlanId}")
+    def paralleId = getParalleId(execName)
+    def buildURL = "${env.BUILD_URL}"
+    def url = generateBluoceanLink("${buildURL}","${paralleId}")
+    echo "URL : " + "${url}"
```
Comment (kasunbg) [inline src/org/wso2/tg/jenkins/executors/TestExecutor.groovy]: Shall we make this log more context aware? For ex. what URL is this?

File: src/org/wso2/tg/jenkins/executors/TestExecutor.groovy [hunk 7]
```diff
@@ -228,3 +271,21 @@ static def getRepositoryBranch(def branch) {
...
+
+/**
+ * Builds the bluocean console link for a given test plan
+ *
+ * @param buildURL Jenkins classic build URL
+ * @param parallelId Parallel id of the test plan
+ * @return Blueocean link
+ */
+def generateBluoceanLink(def buildURL , def parallelId){
+    def split = buildURL.split('/')
```
Comment (kasunbg) [inline src/org/wso2/tg/jenkins/executors/TestExecutor.groovy]: URL split logics are doomed for failure.. Can you derive the url by using jenkine variables? 

See: https://testgrid-live-dev.private.wso2.com/admin/env-vars.html/

You can get jenkins url via **JENKINS_URL**. The _JOB_NAME_ provide a forward-slash separated job name.

File: src/org/wso2/tg/jenkins/executors/TestExecutor.groovy (hunk 1 above)
Comment (sameerawickramasekara) [reply src/org/wso2/tg/jenkins/executors/TestExecutor.groovy]: +1

File: src/org/wso2/tg/jenkins/executors/TestGridExecutor.groovy (hunk 4 above)
Comment (sameerawickramasekara) [reply src/org/wso2/tg/jenkins/executors/TestGridExecutor.groovy]: +1

File: src/org/wso2/tg/jenkins/executors/TestExecutor.groovy (hunk 7 above)
Comment (sameerawickramasekara) [reply src/org/wso2/tg/jenkins/executors/TestExecutor.groovy]: changed the logic to use the Jenkins env variables

**Pull request merged**